src/Compilers/CSharp/Test/Semantic/Semantics/AwaitExpressionTests.cs

```diff
@@ -20,6 +20,97 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     /// </summary>
     public class AwaitExpressionTests : CompilingTestBase
     {
+        [Fact]
+        public void TestAwaitInfoExtensionMethod()
+        {
+            var text =
+@"using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+static class App{
+    public static async Task Main(){
+        MyAwaitable Get()
+        {
+            return new MyAwaitable();
+        }
+
+        var x = Get();
+        x.SetValue(42);
+
+        Console.WriteLine(await x + ""!"");
+    }
+}
+
+struct MyAwaitable
+{
+    private ValueTask<int> task;
+    private TaskCompletionSource<int> source;
+
+    private TaskCompletionSource<int> Source
+    {
+        get
+        {
+            if (source == null)
+            {
+                source = new TaskCompletionSource<int>();
+                task = new ValueTask<int>(source.Task);
+            }
+            return source;
+        }
+    }
+    internal ValueTask<int> Task
+    {
+        get
+        {
+            _ = Source;
+            return task;
+        }
+    }
+
+    public void SetValue(int i)
+    {
+        Source.SetResult(i);
+    }
+}
+
+static class MyAwaitableExtension
+{
+    public static System.Runtime.CompilerServices.ValueTaskAwaiter<int> GetAwaiter(this MyAwaitable a)
+    {
+        return a.Task.GetAwaiter();
+    }
+}";
+            var refApis = System.AppDomain.CurrentDomain.GetAssemblies().Where(a => !a.IsDynamic).Select(
+                a => MetadataReference.CreateFromFile(a.Location)
+            );
+            var tree = SyntaxFactory.SyntaxTree(SyntaxFactory.ParseCompilationUnit(text));
+            var csCompilation = CSharpCompilation.Create("something",
+                   new[] { tree },
+                   refApis,
+                   new CSharpCompilationOptions(OutputKind.ConsoleApplication)
+            );
+
+            var model = csCompilation.GetSemanticModel(tree);
+            var awaitExpression = tree.GetRoot().DescendantNodes().OfType<AwaitExpressionSyntax>().First();
+
+            var op = model.GetOperation(awaitExpression);
```

Comment:
>var op [](http://example.com/codeflow?start=12&length=6)

This local does not appear to be used. If the call to `GetOperation()` is part of the test, consider asserting the returned value is as expected.

Reply:
Oh, I used it just while debugging...